manifests/source.pp

```diff
@@ -1,65 +1,46 @@
 # source.pp
 # add an apt source
 define apt::source(
-  $location          = undef,
-  $comment           = $name,
-  $ensure            = present,
-  $release           = undef,
-  $repos             = 'main',
-  $include           = {},
-  $key               = undef,
-  $pin               = undef,
-  $architecture      = undef,
-  $allow_unsigned    = false,
-  $include_src       = undef,
-  $include_deb       = undef,
-  $required_packages = undef,
-  $key_server        = undef,
-  $key_content       = undef,
-  $key_source        = undef,
-  $trusted_source    = undef,
-  $notify_update     = true,
+  Optional[Variant[String, Stdlib::Compat::String]] $location                         = undef,
+  Variant[String, Stdlib::Compat::String] $comment                                    = $name,
+  Variant[String, Stdlib::Compat::String] $ensure                                     = present,
+  Optional[Variant[String, Stdlib::Compat::String]] $release                          = undef,
+  Variant[String, Stdlib::Compat::String] $repos                                      = 'main',
+  Optional[Variant[Hash, Stdlib::Compat::Hash]] $include                              = {},
+  Optional[Variant[String, Stdlib::Compat::String, Hash, Stdlib::Compat::Hash]] $key  = undef,
+  $pin                                                                                = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $architecture                     = undef,
+  Boolean $allow_unsigned                                                             = false,
+  Boolean $notify_update                                                              = true,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_server                       = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_content                      = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_source                       = undef,
```

Comment:
I was wondering whether or not to leave them in until the second major release. The aim of the first major release is to try not to break current setups - however they were marked as deprecated for the next release - you think I should go ahead and remove them for this release or wait until the second one?